pkgs/by-name/al/alist/update.nix

```diff
@@ -0,0 +1,50 @@
+{
+  writeShellApplication,
+  nix,
+  nix-update,
+  curl,
+  coreutils,
+  gnused,
+  jq,
+  common-updater-scripts,
+}:
+
+writeShellApplication {
+  name = "update-alist";
+  runtimeInputs = [
+    coreutils
+    curl
+    gnused
+    jq
+    nix
+    common-updater-scripts
+    nix-update
+  ];
+
+  text = ''
+    # get old info
+    oldVersion=$(nix-instantiate --eval --strict -A "alist.version" | jq -e -r)
+
+    get_latest_release() {
+        local repo=$1
+        curl --fail ''${GITHUB_TOKEN:+ -H "Authorization: bearer $GITHUB_TOKEN"} \
+             -s "https://api.github.com/repos/AlistGo/$repo/releases/latest" | jq -r ".tag_name"
+    }
+
+    version=$(get_latest_release "alist")
+    version="''${version#v}"
+    webVersion=$(get_latest_release "alist-web")
+
+    if [[ "$oldVersion" == "$version" ]]; then
+        echo "Already up to date!"
+        exit 0
+    fi
+
+    # Using nix-update will cause webVersion to be overwritten.
+    # The solution is to run update-source-version for web after nix-update to ensure that webVersion is correct.
+    nix-update alist --version "$version"
+
+    # The --ignore-same-hash flag bypasses errors caused by the unchanged hash when the URL relies on a non-default version-key.
```

Comment:
I do not think the reason is correct.

Before `nix-update`, `webVersion` is `3.39.2`.  After `nix-update`, `webVersion` is changed to `3.40.0`, which is wrong.  (Maybe open an issue to `nix-update`?)  Then, `update-source-version` changes `webVersion` to its latest version `3.39.2` **which is the same as the original one**, so its hash remains the same.